### ISSUE TITLE
Int2 Serialization breaks after reload

### ISSUE THREAD

**Issue by stefnotch** [FlaxEngine/FlaxEngine#375]
**Issue description:**
I have a [json asset](https://docs.flaxengine.com/manual/scripting/tutorials/custom-asset.html) with an `Int2`. This works flawlessly, until I change any script and then recompile the scripts.

After that, the following error gets thrown
```
[ 00:04:34.680 ]: [Warning] Exception has been thrown during . Object reference not set to an instance of an object
Stack strace:
  at (wrapper dynamic-method) FlaxEngine.Int2.CreateFlaxEngine.Int2()
  at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject (Newtonsoft.Json.JsonReader reader, System.Type objectType, Newtonsoft.Json.Serialization.JsonContract contract, Newtonsoft.Json.Serialization.JsonProperty member, Newtonsoft.Json.Serialization.JsonContainerContract containerContract, Newtonsoft.Json.Serialization.JsonProperty containerMember, System.Object existingValue) <0x1e293ca8a90 + 0x00d8a> in <3363f129c16943e0a230d689dd1a1295>:0 
  at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal (Newtonsoft.Json.JsonReader reader, System.Type objectType, Newtonsoft.Json.Serialization.JsonContract contract, Newtonsoft.Json.Serialization.JsonProperty member, Newtonsoft.Json.Serialization.JsonContainerContract containerContract, Newtonsoft.Json.Serialization.JsonProperty containerMember, System.Object existingValue) <0x1e293ca7bd0 + 0x00262> in <3363f129c16943e0a230d689dd1a1295>:0 
  at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue (Newtonsoft.Json.Serialization.JsonProperty property, Newtonsoft.Json.JsonConverter propertyConverter, Newtonsoft.Json.Serialization.JsonContainerContract containerContract, Newtonsoft.Json.Serialization.JsonProperty containerProperty, Newtonsoft.Json.JsonReader reader, System.Object target) <0x1e293cd1280 + 0x005aa> in <3363f129c16943e0a230d689dd1a1295>:0 
  at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject (System.Object newObject, Newtonsoft.Json.JsonReader reader, Newtonsoft.Json.Serialization.JsonObjectContract contract, Newtonsoft.Json.Serialization.JsonProperty member, System.String id) <0x1e293cb25d0 + 0x018ba> in <3363f129c16943e0a230d689dd1a1295>:0 
  at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Populate (Newtonsoft.Json.JsonReader reader, System.Object target) <0x1e2920ceae0 + 0x00d2a> in <3363f129c16943e0a230d689dd1a1295>:0 
  at Newtonsoft.Json.JsonSerializer.PopulateInternal (Newtonsoft.Json.JsonReader reader, System.Object target) <0x1e2920ce6d0 + 0x0028a> in <3363f129c16943e0a230d689dd1a1295>:0 
  at Newtonsoft.Json.JsonSerializer.Populate (Newtonsoft.Json.JsonReader reader, System.Object target) <0x1e2920ce5e0 + 0x00082> in <3363f129c16943e0a230d689dd1a1295>:0 
  at FlaxEngine.Json.JsonSerializer.Deserialize (System.Object input, System.String json) [0x0002c] in C:\Users\Stefnotch\Documents\GitHub\Flax\FlaxEngine\Source\Engine\Serialization\JsonSerializer.cs:434 
  at FlaxEngine.JsonAsset.CreateInstance () [0x00076] in C:\Users\Stefnotch\Documents\GitHub\Flax\FlaxEngine\Source\Engine\Content\JsonAsset.cs:47 
[ 00:04:34.680 ]: [Warning] Exception has been thrown during .
Object reference not set to an instance of an object

```

**Steps to reproduce:**
1. Write the following code
```cs
public class BreaksStuff
{
    public Int2 SomeValue { get; set; } = new Int2(3, 5);
}
```
2. Go to the `Content` folder and create a JSON asset with that type
3. Recompile scripts
4. Try to open the JSON asset

**Flax version:**
Flax 1.1, latest from master

**Comment by mafiesto4** [comment]
I was unable to reproduce it. Does it contain any more info? Like nested exception above this log?

**Comment by stefnotch** [comment]
Okay, I whipped up a minimal project that should reproduce this issue. It's a bit finicky to reproduce sometimes. 
The log doesn't seem to contain more info (see: attached project)


[Int2Ser.zip](https://github.com/FlaxEngine/FlaxEngine/files/6185015/Int2Ser.zip)


![image](https://user-images.githubusercontent.com/10220080/112052174-1c672400-8b53-11eb-8d01-ec46ef647090.png)

**Comment by mafiesto4** [comment]
The problem was related to reloading scripts without any scenes loaded. In this case the faster path was picked and some events were not triggered. Fixed in fcd86bccf5841ab8027b7747b53ee2b2ed8d1d87.

*note that editor auto closes custom Json assets editor windows as we're reloading code - could be improved by restoring those windows later (todo).
